client/components/tinymce/plugins/mentions/suggestion.jsx

```diff
@@ -0,0 +1,69 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+const getRegExpFor = function( type, textToHighlight ) {
+	const expressions = {};
+
+	expressions.username = '(^' + textToHighlight + ')(\\w*)\\s*';
+	expressions.fullName = '(^.*?)(\\b' + textToHighlight + ')(.*)';
+
+	return new RegExp( expressions[ type ], 'ig' );
+};
+
+const highlight = function( content, textToHighlight, type ) {
+	const matcher = getRegExpFor( type, textToHighlight ),
+		matches = matcher.exec( content );
+
+	if ( matches ) {
+		const highlights = [];
+		let highlighted = false;
+
+		for ( let i = 1, length = matches.length; i < length; i++ ) {
```

Comment:
Maybe an `Array#map` would be suitable here?

```js
return matches.map( ( item, i ) => {
	if ( textToHighlight.toLowerCase() === item.toLowerCase() ) {
		return <strong className="mentions__highlight" key={ i }>{ item }</strong>;
	}

	return item;
} );
```

Can you explain the need for your highlighted boolean?